config/packages/dev/mailer.yaml

```diff
@@ -0,0 +1,5 @@
+framework:
+    mailer:
+        envelope:
+            # TODO this does not seem to work
+            recipients: ['%env(APP_DEV_EMAIL_RECIPIENT)%']
```

Comment:
Anyone has a clue here why this would fail?  I have the env var set in .env.local, and yet emails seem to go wherever they want.

Reply:
This goes far beyond my understanding of the Mailer component, but...

With Mailtrap set up on SMTP, inside the `SmtpTransport` class when I dump [the recipients](https://github.com/symfony/mailer/blob/v4.4.17/Transport/Smtp/SmtpTransport.php#L198-L200) I get the right address (the one from this config), but when I dump [the message stream](https://github.com/symfony/mailer/blob/v4.4.17/Transport/Smtp/SmtpTransport.php#L203-L205) the body it builds has the original email address added to the email.  That's about as far as I can get before I'm over my head.

Reply:
Emails have 2 different concepts: the recipients in the mime messages (the `To` and `Cc` headers) and the recipients in the SMTP protocol (which is where the email gets routed by SMTP).
When using this `recipients` feature, the Mailer component overrides things only at the level of the SMTP envelope, not in the mime headers. So you will still see the original recipients in the headers in your email client, even though emails should be delivered to your own mailbox.

Reply:
Ah that may explain the problem I saw then, as I tested it with a foo@ alias which lands in my inbox anyway.. Will test overriding with an other domain entirely.

Reply:
you could look at the `Delivered-To` headers in the received email (which is how these SMTP-level features are surfaced)